### ISSUE TITLE
AOT - Calling function 'ɵb', function calls are not supported.

### ISSUE THREAD

**Issue by kukjevov** [angular/angular#15571]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
ERROR in Error encountered resolving symbol values statically. Calling function 'ɵb', function calls are not supported. Consider replacing the function or lambda with a reference to an exported function, resolving symbol HostBinding in d:/data/git/universal-demo/node_modules/@angular/core/core.d.ts, resolving symbol BaseAnimatedComponent in d:/data/git/universal-demo/app/misc/baseAnimatedComponent.ts, resolving symbol BaseAnimatedComponent in d:/data/git/universal-demo/app/misc/baseAnimatedComponent.ts

**Expected behavior**
It works without AOT. With AOT it works if i remove `@HostBinding()` from base class and use it in inherited class it works ok.

**Minimal reproduction of the problem with instructions**
```typescript
import {Component, OnInit, HostBinding} from '@angular/core';
import {DataService} from "../../services/api/data/data.service";
import { BaseAnimatedComponent } from "app/misc/baseAnimatedComponent";
import { trigger, transition, animate, style } from "@angular/animations";

/**
 * Home component
 */
@Component(
{
    selector: 'home-view',
    templateUrl: 'home.component.html',
    providers: [DataService],
    animations: 
    [
        trigger('flyInOut',
        [
            transition('void => *', 
            [
                style(
                {
                    opacity: 0,
                    transform: 'translateX(-20%)'
                }),
                animate('400ms ease-in', style(
                {
                    opacity: 1,
                    transform: 'translateX(0)'
                }))
            ]),
            transition('* => void', 
            [
                animate('400ms ease-out', style(
                {
                    opacity: 0,
                    transform: 'translateX(20%)'
                }))
            ])
        ])
    ]
})
export class HomeComponent extends BaseAnimatedComponent implements OnInit
{
public static ngRoutes = [{path: '', component: HomeComponent}];
    //######################### public properties #########################
    public subs: string;
    public counter = 0;

    //######################### public properties - bindings #########################

    /**
     * Attach animation directly to component (enter, exit)
     */
    @HostBinding('@flyInOut') 
    public animatedComponent = true;

    //######################### constructor #########################
    constructor(private dataSvc: DataService)
    {
        super();
    }

    //######################### public methods #########################
    public ngOnInit()
    {
        this.dataSvc.getData().map(data =>
        {
            return `${data.greeting} ${data.name}`;
        }).subscribe(data =>
        {
            this.subs = data;
        });
    }

    public inc()
    {
        this.counter++;
    }
}
```

Base class. If i comment that `@HostBinding` it works.
```typescript
import {HostBinding, HostListener} from '@angular/core';
import {AnimationEvent} from '@angular/animations';

/**
 * Base class that enables fly in out animation
 */
export class BaseAnimatedComponent
{
    //######################### public properties - bindings #########################

    /**
     * Attach animation directly to component (enter, exit)
     */
    @HostBinding('@flyInOut') 
    public animatedComponent = true;
}
```

**What is the motivation / use case for changing the behavior?**

**Please tell us about your environment:**
Windows 10, VSCode

* **Angular version:** 4.0.0

* **Browser:** [all]
 
* **Language:** [TypeScript 2.2.1]

* **Node (for AoT issues):** `node --version` = v6.9.4

**Comment by kukjevov** [comment]
Also interesting thing is that if i add `@Injectable()` to base class `BaseAnimatedComponent` AOT will fail with different error. 

```
ERROR in Error encountered resolving symbol values statically. Calling function 'ɵmakeDecorator', function calls are not supported. Consider replacing the function or lambda with a reference to an exported function, resolving symbol Injectable in d:/data/git/universal-demo/node_modules/@angular/core/core.d.ts, resolving symbol BaseAnimatedComponent in d:/data/git/universal-demo/app/misc/baseAnimatedComponent.ts, resolving symbol BaseAnimatedComponent in d:/data/git/universal-demo/app/misc/baseAnimatedComponent.ts
```

This is something i noticed also in other class that extends base class. Interesting thing is that this does not happen every time. I extend class from package in 2 different projects. In one i get mentioned error and in second it works. So it is quite hard to simulate sometimes.

Is there any way how to provide more debug information? 

One more question is there need to set `@Injectable()` for base class if it has constructor with DI tokens and child class does not have any constructor, so DI uses base class constructor? 

By the way i have notices commit https://github.com/angular/angular/commit/8ef621ad2ad5662de56af1b5eb8ed2bff45809c2 could this solve this issue?

**Comment by kukjevov** [comment]
I have created repository where you can see described problem with `eb` error.

https://github.com/kukjevov/ng-universal-demo/tree/eberror

Its in branch `eberror`.
run `npm install`
then `npm run build:aot` or `npm run build:min`.

Is there anyone looking at this at least? Or it is low importance?

So far as i can see AOT is great idea, but cant be used in real project.

Thank you anyway :)

**Comment by DzmitryShylovich** [comment]
> Calling function 'ɵmakeDecorator', function calls are not supported

usually this means that something wrong with 3rd party dependencies.

**Comment by kukjevov** [comment]
Well usually maybe, but it looks like not this case. Because i have added `@Injectable()` to or even `Component` and this error occurs. What is even worse description from error message does not help to solve problem. There is no 3rd party code except @ngx-translate, but i think if i remove it it will throw same error, because there is not used this.

I created issue showing second error https://github.com/angular/angular/issues/15767.

**Comment by kukjevov** [comment]
I am closing this, since it was caused by using absolute paths which will be solved with other issue #15767
